projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/graphql/strategy/datajob/JobFieldStrategyByDataJobContacts.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2021 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.vmware.taurus.service.graphql.strategy.datajob;
+
+import com.vmware.taurus.controlplane.model.data.DataJobContacts;
+import com.vmware.taurus.service.graphql.model.Criteria;
+import com.vmware.taurus.service.graphql.model.Filter;
+import com.vmware.taurus.service.graphql.model.V2DataJob;
+import com.vmware.taurus.service.graphql.strategy.FieldStrategy;
+import org.apache.commons.collections.CollectionUtils;
+
+import java.util.Comparator;
+import java.util.function.Predicate;
+
+public class JobFieldStrategyByDataJobContacts extends FieldStrategy<V2DataJob> {
+   private static final Comparator<V2DataJob> COMPARATOR_DEFAULT = Comparator.comparing(e -> e.getConfig().getContacts(),
+         Comparator.nullsLast((o1, o2) -> {
+            boolean o1ContactPresent = isContactPresent(o1);
+            boolean o2ContactPresent = isContactPresent(o2);
+            return Boolean.compare(o1ContactPresent, o2ContactPresent);
+         }));
+
+   @Override
+   public JobFieldStrategyBy getStrategyName() {
+      return JobFieldStrategyBy.DATA_JOB_CONTACTS;
+   }
+
+   @Override
+   public Criteria<V2DataJob> computeFilterCriteria(Criteria<V2DataJob> criteria, Filter filter) {
+      var predicate = criteria.getPredicate();
```

Comment:
It would be good if validate the filter as it is not supported. You can take a look at JobFieldStrategyByNextRun for reference.

Reply:
Added a test